### ISSUE TITLE
Setup correct RBAC roles/permissions for ops-user in VCH (--ops-user)

### ISSUE THREAD

**Issue by lcastellano** [vmware/vic#6492]
**User Statement:**
The VCH must perform operations in VCenter (e.g. create/delete users). In VIC 1.2 the creator of the VCH has two options:

1. Supply the VCenter administrator credentials,
2. Create and manually configure another VCenter user with minimum roles and permission.

While the first alternative wins on usability it does not provide the adequate barriers among VCHs. In addition, it suffers from security weaknesses since each VCH can act as the full VCenter administration. The second approach requires a lot of tedious and error prone manual configuration in VSphere, but it provides the needed security and isolation.
The goal of this epic is to automate the creation/configuration of the VCenter user (ops-user) described by (2).

**Details:**
Automation of creation and configuration of the VIC user should be performed by vic-machine (or vic-machine-service) and implemented as a set of APIs. The are three main tasks:

1. Creation of ops-user
2. Creation of the roles that ops-user will have, this has to be done only once in VCenter,
3. Assignment of permissions for the different VCenter resources managed by the VCH. Some of the resources are created by vic-machine and therefore permission assignment must be done at the end of VCH creation.
4. Validation of the roles and permission given and existing VCH and an exiting ops-user.

Since creation of the user is not currently supported in go-vmomi, we believe that manual creation of the user + automated Role/Permission assignment and Validation of the ops-user should be a valuable intermediate goal.

**Acceptance Criteria:**
Design and implementation of the feature described above. The first implementation will skip the creation of the ops-User.

**Comment by lcastellano** [comment]
This is from issue: https://github.com/vmware/vic/issues/1689

Dumping the RBAC config from a working VCH deployed on vCenter 6.5:

```
vagrant@devbox:~$ for i in "VCH - vcenter" "VCH - datacenter" "VCH - datastore" "VCH - network" "VCH - endpoint"; do  echo ======= Role $i ============;echo "******" Configuration; govc role.ls "$i"; echo;echo "******" Usage; govc role.usage "$i"; echo;echo;done  | grep -v '^System\.'
======= Role VCH - vcenter ============
****** Configuration
Datastore.Config

****** Usage
Role           Entity  Principal          Propagate
VCH - vcenter  /       VSPHERE.LOCAL\ops  No


======= Role VCH - datacenter ============
****** Configuration
Datastore.Config
Datastore.FileManagement

****** Usage
Role              Entity  Principal          Propagate
VCH - datacenter  /dc1    VSPHERE.LOCAL\ops  No


======= Role VCH - datastore ============
****** Configuration
Datastore.AllocateSpace
Datastore.Browse
Datastore.Config
Datastore.DeleteFile
Datastore.FileManagement
Host.Config.SystemManagement

****** Usage
Role             Entity                        Principal          Propagate
VCH - datastore  /dc1/datastore/vsanDatastore  VSPHERE.LOCAL\ops  No
VCH - datastore  /dc1/host/cluster1            VSPHERE.LOCAL\ops  Yes


======= Role VCH - network ============
****** Configuration
Network.Assign

****** Usage
Role           Entity                                  Principal          Propagate
VCH - network  /dc1/network/netFolder/InternalNetwork  VSPHERE.LOCAL\ops  No


======= Role VCH - endpoint ============
****** Configuration
DVPortgroup.Create
DVPortgroup.Delete
DVPortgroup.Modify
DVPortgroup.PolicyOp
DVPortgroup.ScopeOp
VApp.AssignVM
VirtualMachine.Config.AddNewDisk
VirtualMachine.Config.AdvancedConfig
VirtualMachine.Config.EditDevice
VirtualMachine.Config.RemoveDisk
VirtualMachine.GuestOperations.Execute
VirtualMachine.Interact.DeviceConnection
VirtualMachine.Interact.PowerOff
VirtualMachine.Interact.PowerOn
VirtualMachine.Inventory.Create
VirtualMachine.Inventory.Delete
VirtualMachine.Inventory.Register
VirtualMachine.Inventory.Unregister

****** Usage
Role            Entity                                     Principal          Propagate
VCH - endpoint  /dc1/host/cluster1/Resources/ghicken-vc65  VSPHERE.LOCAL\ops  Yes
```

Additionally, `Read-Only, with Propagate to Children` is required on the parent of any Distributed vSwitch's that are used by the VCH. To avoid adding Read-Only to the datacenter, you may create a network inventory folder and place the DVS in that.

The `VCH - endpoint` configuration should be applied to the vApp that is deployed and should propagate.

**Update**
The dvPortGroup.Create|Delete are not needed for ongoing operations.

**Comment by lcastellano** [comment]
This is from Doug

Next steps would be in vic-machine:
- We can create a new role using govmomi object.AuthorizationManager.AddRole (see govc/role/create.go for reference)
- Role privilege set still TBD, but can be limited to specific entities in the step below (vs all managed entities as the Admin role is).  Note that the full privilege set can be viewed using `govc role.ls Admin`
- Use govmomi object.AuthorizationManager.SetEntityPermissions to grant principal (-op-username) role permission on the VCH vapp/resource-pool entity (see govc/permission/set.go for reference)

**Comment by mdubya66** [comment]
All issues are closed. Closing the epic
